src/main/java/com/chatcode/service/ScrapService.java

```diff
@@ -0,0 +1,93 @@
+package com.chatcode.service;
+
+import com.chatcode.domain.entity.Article;
+import com.chatcode.domain.entity.Avatar;
+import com.chatcode.domain.entity.scrap.Scrap;
+import com.chatcode.dto.scrap.ScrapRequestDTO;
+import com.chatcode.dto.scrap.ScrapResponseDTO;
+import com.chatcode.exception.common.ContentNotFoundException;
+import com.chatcode.repository.article.ArticleReadRepository;
+import com.chatcode.repository.avatar.AvatarReadRepository;
+import com.chatcode.repository.scrap.ScrapReadRepository;
+import com.chatcode.repository.scrap.ScrapWriteRepository;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+@RequiredArgsConstructor
+@Service
+public class ScrapService {
+    private final ScrapWriteRepository scrapWriteRepository;
+    private final ScrapReadRepository scrapReadRepository;
+    private final ArticleReadRepository articleReadRepository;
+    private final AvatarReadRepository avatarReadRepository;
+
+    @Transactional
+    public ScrapResponseDTO scrap(ScrapRequestDTO scrapRequestDTO) {
+        LocalDateTime now = LocalDateTime.now();
+
+        Optional<Avatar> optionalAvatar = avatarReadRepository.findById(scrapRequestDTO.getAvatarId());
+        Avatar avatar = optionalAvatar.orElseThrow(() -> new ContentNotFoundException("Avatar not found"));
```

Comment:
```suggestion
        Avatar avatar = avatarReadRepository.findById(scrapRequestDTO.getAvatarId())
                        .orElseThrow(() -> new ContentNotFoundException("Avatar not found"));
```
Optional은 요롷게 사용도 할 수 있습니다!!

Reply:
참고해서 수정해보겠습니다 ~!!

Reply:
오 훨씬 간결해지네욥